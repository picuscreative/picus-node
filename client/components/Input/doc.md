
## Input

From [`Input`](Input)

Input:
Component that renders a text inside an h1,
intended to be used as a title

prop | type | default | required | description
---- | :----: | :-------: | :--------: | -----------
**alignCenter** | `Boolean` |  | :x: | Allows the alignment of the element to the center
**className** | `String` |  | :x: | Class name for the container
**error** | `Boolean` |  | :x: | Change style if input has an error condition
**handleChange** | `Function` |  | :x: | Value state setter
**name** | `String` |  | :white_check_mark: | Name to be given to identify the input element
**pattern** | `String` |  | :x: | Pattern for validating in case of HTML5
**placeholder** | `String` |  | :x: | Text that will appear in the element
**type** | `String` |  | :x: | Definition of the input type
**value** | `String` |  | :x: | Value to be given to the input element
**withPadding** | `Boolean` |  | :x: | Allows to add a fixed padding
**withPaddingTop** | `Boolean` |  | :x: | Allows to add a fixed padding top



