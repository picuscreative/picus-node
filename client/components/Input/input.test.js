/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import Input from '~/components/Input';

describe('Input', () => {
  it('should render an empty input', () => {
    const component = shallow(<Input name="test" />);

    expect(component.getElements()).toMatchSnapshot();
  });
});
