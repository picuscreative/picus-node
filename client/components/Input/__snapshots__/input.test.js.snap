// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render an empty input 1`] = `
Array [
  <input
    className="input"
    name="test"
    type="text"
  />,
]
`;
